core/src/main/java/org/elasticsearch/search/fetch/subphase/highlight/CustomQueryScorer.java

```diff
@@ -94,5 +93,13 @@ protected void extract(Query query, float boost, Map<String, WeightedSpanTerm> t
                 super.extract(query, boost, terms);
             }
         }
+
+        /**
+         * Workaround to detect parent/child query
+         */
+        private static final String PARENT_CHILD_QUERY_NAME = "HasChildQueryBuilder$LateParsingQuery";
+        private static boolean isChildOrParentQuery(Class<?> clazz) {
```

Comment:
We have a test for this in `ChildQuerySearchIT#testHighlighHighlightersIgnoreParentChild`